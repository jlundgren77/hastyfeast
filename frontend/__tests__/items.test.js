import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'ABC123',
  title: 'A test item',
  price: 5000,
  description: 'This items is for a test',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg',
};

describe('<Item />', () => {
  it('renders and matches snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  //   it('renders and displays properly', () => {
  //     const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //     const PriceTag = wrapper.find('PriceTag');

  //     expect(PriceTag.children().text()).toBe('$50');
  //     expect(wrapper.find('Title a').text()).toBe('A test item');
  //     const image = wrapper.find('img');
  //     expect(image.props().src).toBe(fakeItem.image);
  //     expect(image.props().alt).toBe(fakeItem.title);
  //   });

  //   it('renders out the buttons', () => {
  //     const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //     const buttonList = wrapper.find('.buttonList');
  //     expect(buttonList.children()).toHaveLength(3);
  //     expect(buttonList.find('AddToCart').exists()).toBeTruthy();
  //     expect(wrapper.find('AddToCart').props().id).toBe(fakeItem.id);
  //   });
});
