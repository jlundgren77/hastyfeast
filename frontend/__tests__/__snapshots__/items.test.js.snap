// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A test item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A test item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This items is for a test
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
